crates/noirc_errors/src/reporter.rs

```diff
@@ -115,6 +115,10 @@ pub fn report_all(
     diagnostics: &[FileDiagnostic],
     deny_warnings: bool,
 ) -> ReportedErrors {
+    // Report warnings before any errors
+    let (mut diagnostics, mut errors): (Vec<_>, _) = diagnostics.into_iter()
+        .partition(|item| item.diagnostic.is_warning());
+    diagnostics.append(&mut errors);
```

Comment:
Wouldn't it be better to implement this using a sorting rule rather than allocating two new vectors and then appending one onto the other?

Reply:
It'd be ideal if `partition_in_place` were stabilized

Reply:
Yeah, I doubt that this will cause too much overhead in practice (If you've got that many errors/warnings in your project then you've got bigger problems)